### PR TITLE
[Docs] Explain limitations when a custom graph is a source of the filter interaction

### PR DESCRIPTION
**Pull request by petar-qb**
## Description
Solves this https://github.com/McK-Internal/vizro-internal/issues/1101 Issue.

Also reference to the original Issue -> https://github.com/McK-Internal/vizro-internal/issues/956

## Screenshot

Exception raised at run time. 
image

Docs changes:
image

## Notice

- [x] I acknowledge and agree that, by checking this box and clicking "Submit Pull Request":

  - I submit this contribution under the [Apache 2.0 license](https://www.apache.org/licenses/LICENSE-2.0.txt) and represent that I am entitled to do so on behalf of myself, my employer, or relevant third parties, as applicable.
  - I certify that (a) this contribution is my original creation and / or (b) to the extent it is not my original creation, I am authorized to submit this contribution on behalf of the original creator(s) or their licensees.
  - I certify that the use of this contribution as authorized by the Apache 2.0 license does not violate the intellectual property rights of anyone else.
  - I have not referenced individuals, products or companies in any commits, directly or indirectly.
  - I have not added data or restricted code in any commits, directly or indirectly.

### CODE REVIEW

File: vizro-core/docs/pages/user-guides/actions.md [hunk 1]
```diff
@@ -217,6 +217,23 @@ Here is an example of how to configure a chart interaction when the source is a
 
     [Graph2]: ../../assets/user_guides/actions/actions_filter_interaction.png
 
+
+!!! note "`filter_interaction with custom charts`"
```
Comment (maxschulz-COL) [inline vizro-core/docs/pages/user-guides/actions.md:221]:
```suggestion
!!! note "`filter_interaction` with custom charts"
```

File: vizro-core/docs/pages/user-guides/actions.md [hunk 2]
```diff
@@ -217,6 +217,23 @@ Here is an example of how to configure a chart interaction when the source is a
...
+
+    If `filter_interaction` is assigned to the [custom chart](custom-charts.md), ensure that `custom_data` is passed from the `vm.Graph` figure to the `@capture('graph')` function, and then assigned to the returned figure object.
```
Comment (maxschulz-COL) [inline vizro-core/docs/pages/user-guides/actions.md:223]:
```suggestion
    If `filter_interaction` is assigned to a [custom chart](custom-charts.md), ensure that `custom_data` is an argument of the custom chart function, and that the relevant entry is then passed to the underlying plotly function. When configuring the custom chart in `vm.Graph`, ensure that `custom_data` is passed.
```

File: vizro-core/src/vizro/models/_components/graph.py [hunk 3]
```diff
@@ -102,7 +102,12 @@ def _filter_interaction(
         try:
             custom_data_columns = model_manager[source_graph_id]["custom_data"]
         except KeyError as exc:
-            raise KeyError(f"No `custom_data` argument found for source graph with id {source_graph_id}.") from exc
+            raise KeyError(
+                f"Missing 'custom_data' for the source graph with id {source_graph_id}. "
```
Comment (maxschulz-COL) [inline vizro-core/src/vizro/models/_components/graph.py:106]: And maybe something similar as my suggestion above?

I overall feel we need to make clear three things:
1. have a `custom_data` argument
2. pass the input to the underlying plotly function
3. fill it in when configuring the graph

Maybe my suggestion helps, or a combination of the two?

Comment (petar-qb) [reply vizro-core/src/vizro/models/_components/graph.py:106]: I updated the PR description with the latest changes. Do they look ok to you?

**Review by stichbury — APPROVED** [review]
LGTM from docs perspective 🏆

File: vizro-core/docs/pages/user-guides/actions.md [hunk 4]
```diff
@@ -217,6 +217,24 @@ Here is an example of how to configure a chart interaction when the source is a
 
     [Graph2]: ../../assets/user_guides/actions/actions_filter_interaction.png
 
+
+!!! note "`filter_interaction` with custom charts"
+
+    If `filter_interaction` is assigned to a [custom chart](custom-charts.md), ensure that `custom_data` is an argument of the custom chart function, and that the relevant entry is then passed to the underlying plotly function.
```
Comment (maxschulz-COL) [inline vizro-core/docs/pages/user-guides/actions.md:223]:
```suggestion
    If `filter_interaction` is assigned to a [custom chart](custom-charts.md), ensure that `custom_data` is an argument of the custom chart function, and that this argument is then passed to the underlying plotly function.
```

File: vizro-core/docs/pages/user-guides/actions.md [hunk 5]
```diff
@@ -217,6 +217,24 @@ Here is an example of how to configure a chart interaction when the source is a
...
+    When configuring the custom chart in `vm.Graph`, ensure that `custom_data` is passed.
```
Comment (maxschulz-COL) [inline vizro-core/docs/pages/user-guides/actions.md:224]:
```suggestion
    When then adding the custom chart in `vm.Graph`, ensure that `custom_data` is passed.
```

**Review by maxschulz-COL — APPROVED** [review]
Looks good, I had a slight change of heart about my own suggestion. Wdyt? If you agree, would also change it in the error message, but good to go

**Pull request merged**